src/sugar3/activity/widgets.py

```diff
@@ -396,3 +396,31 @@ def __init__(self):
         self.paste = PasteButton()
         self.insert(self.paste, -1)
         self.paste.show()
+
+class CollabTextEditor(Gtk.Box):
+    def __init__(self,activity):
+        Gtk.Box.__init__(self)
+        self._collab = CollabWrapper(activity)
+        self._collab.connect('message', self.__message_cb)
+        self._collab.setup()
+        self.textview = Gtk.TextView()
+        self.textview.set_editable(True)
+        self.textview.set_cursor_visible(True)
+        self.textview.set_wrap_mode(Gtk.WrapMode.WORD)
+        self.textbuffer = self.textview.get_buffer()
+        self.textbuffer.connect('changed', self.__text_buffer_edited_cb)
+        self.textbuffer.set_text("TESTING")
+        self.textview.show()
+        self.pack_start(self.textview, expand=True, fill=True, padding=0)
+        self.show()
+
+    def __message_cb(self, collab, buddy, message):
+        action = message.get('action')
+	if action == 'entry_changed':
```

Comment:
I think you should let the activity assign the id.

On Mon, Dec 14, 2015, 4:09 PM Batchu Venkat Vishal <notifications@github.com>
wrote:

> In src/sugar3/activity/widgets.py
> <https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47461956>
> :
>
> > +        self._collab.connect('message', self.__message_cb)
> > +        self._collab.setup()
> > +        self.textview = Gtk.TextView()
> > +        self.textview.set_editable(True)
> > +        self.textview.set_cursor_visible(True)
> > +        self.textview.set_wrap_mode(
>
> Gtk.WrapMode.WORD)
> > +        self.textbuffer = self.textview.get_buffer()
> > +        self.textbuffer.connect('changed', self.__text_buffer_edited_cb)
> > +        self.textbuffer.set_text("TESTING")
> > +        self.textview.show()
> > +        self.pack_start(self.textview, expand=True, fill=True, padding=0)
> > +        self.show()
> > +
> > +    def __message_cb(self, collab, buddy, message):
> > +        action = message.get('action')
> > +	if action == 'entry_changed':
>
> Should I randomly generate the ID? Or is there a better way to assign ID's?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274/files#r47461956>
> .
>